synapse/cores/common.py

```diff
@@ -1387,31 +1392,21 @@ def _bumpTufoCache(self, tufo, prop, oldv, newv):
                 if atlim:
                     self._delCacheKey(ckey)
 
-    def _onSetAxonUrl(self, url):
-        old_axon = None
-        if self.axon:
-            old_axon = self.axon
-
-        proxy = s_telepath.openurl(url)
-        reflections = s_reflect.getItemInfo(proxy)
-        classes = reflections.get('inherits')
-        if 'synapse.axon.Axon' in classes:
-            self.axon = proxy
-        elif 'synapse.lib.service.SvcBus' in classes:
-            proxy.fini()
-            # This is a delayed import to avoid a startup import loop
-            import synapse.axon as s_axon
-            svcprox = s_service.openurl(url)
-            self.axon = s_axon.AxonCluster(svcprox)
-        else:  # pragma: no cover
-            proxy.fini()
-            # This scenario is a no-op and the exception is raised
-            # to the eventbus and logged.
-            mesg = 'axon:url must point to a standalone Axon or a ServiceBus'
-            raise s_common.BadConfValu(mesg=mesg, valu=url, key='axon:url')
-
-        if old_axon:
-            old_axon.fini()
+    def _onSetAxonConf(self, conf):
+        auth = conf.get('auth')
+        if not auth:
+            raise s_common.BadConfValu(mesg='auth must be set', valu=conf, key='axon:conf')
+
+        addr = (conf.get('host'), conf.get('port'))
+        if not all(part is not None for part in addr):
+            raise s_common.BadConfValu(mesg='host and port must be set', valu=conf, key='axon:conf')
+
+        self.axon_user = s_neuron.CellUser(auth)
+        self.axon_sess = self.axon_user.open(addr, timeout=30)
```

Comment:
What happens if the session becomes fini'd? Maybe instead we want to be checking that the core *should* have an axonclient, and reconnecting if it gets disconnected?